api/main.py

```diff
@@ -116,25 +117,38 @@ async def get_user(user: User = Depends(get_current_user)):
     return user
 
 
-@app.post('/user/{username}', response_model=User)
+@app.post('/user/{username}', response_model=User,
+          response_model_by_alias=False)
 async def post_user(
         username: str, password: Password, is_admin: bool = False,
+        groups: List[str] = Query([]),
```

Comment:
Why not just use the `User` model actually?  Maybe there was a reason for handling the fields by hand but I don't remember right now.

Reply:
Maybe because `User` model takes a hashed password and we would get plain text password in the request?

Reply:
Maybe, but in both cases it's just a `str` afaik.  I'll think about it again to get to the bottom of it, but that's not a blocking issue for this PR anyway.  It's just whether we can simplify or consolidate the code somehow.

Reply:
OK I think we can merge this now as it's not a new thing anyway, and yes the User model fields are handled individually due to the hashed password so that's all fine.  We might still be able to consolidate things by replacing the password string with a hash before storing the object in the database, to avoid detailing all the fields in the function arguments.  But that's a grey area and just some potential clean-up.